### PR TITLE
sorqa85

### PR DESCRIPTION
**Pull request by pk-sgent**
image

### CODE REVIEW

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/entries/CreateNewTravelEntryPage.java [hunk 1]
```diff
@@ -39,4 +39,11 @@
   public static final By POINT_OF_ENTRY_DETAILS_INPUT =
       By.cssSelector(".v-window #pointOfEntryDetails");
   public static final By SAVE_BUTTON = By.cssSelector(".v-window #commit");
+  public static final By PICK_OR_CREATE_PERSON_TITLE_DE =
+      By.xpath("//div[@class='v-window-header']");
+  public static final By SAVE_POPUP_CONTENT = By.cssSelector(".popupContent #commit");
+  public static final By PICK_A_EXISTING_PERSON_LABEL_DE =
+      By.xpath("//*[text()='W\u00E4hlen Sie eine passende Person']");
+  public static final By PICK_A_EXISTING_CASE_LABEL_DE =
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/entries/CreateNewTravelEntryPage.java:47]: isn't the EN version working as well for this? to have a common ground for them

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/TravelEntryService.java [hunk 2]
```diff
@@ -62,4 +62,23 @@ public TravelEntry buildGeneratedEntryDE() {
         .pointOfEntryDetails("Automated test dummy description")
         .build();
   }
+
+  public TravelEntry buildGeneratedEntryWithPointOfEntryDetails(String pointOfEntry) {
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/TravelEntryService.java:66]: mention also that is for DE to differenciate them

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java [hunk 3]
```diff
@@ -63,6 +69,11 @@
   public static TravelEntry aTravelEntry;
   public static TravelEntry newCaseFromTravelEntryData;
   public static Case aCase;
+  String firstName;
+  String lastName;
+  String sex;
+  String disease;
+  String entryPoint = "Test entry point";
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:76]: why having this data here?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java [hunk 4]
```diff
@@ -216,7 +249,7 @@ public CreateNewTravelEntrySteps(
         });
 
     When(
-        "I check if first and last for case in travel entry is correct",
+        "I check if first and last user name for case in travel entry is correct",
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:252]: user?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java [hunk 5]
```diff
@@ -226,6 +259,58 @@ public CreateNewTravelEntrySteps(
...
+          String checkPopupTitle =
+              webDriverHelpers
+                  .getTextFromWebElement(PICK_OR_CREATE_PERSON_TITLE_DE)
+                  .toLowerCase(Locale.GERMAN);
+          softly.assertEquals(
+              checkPopupTitle,
+              expectedTitle.toLowerCase(Locale.GERMAN),
+              "Wrong popup title for Pick or create person");
+          softly.assertAll();
+        });
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:278]: not sure what to say about this method. Can we make it generic? to work for both DE and EN?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java [hunk 6]
```diff
@@ -226,6 +259,58 @@ public CreateNewTravelEntrySteps(
...
+              webDriverHelpers
+                  .getTextFromWebElement(PICK_OR_CREATE_PERSON_TITLE_DE)
+                  .toLowerCase(Locale.GERMAN);
+          softly.assertEquals(
+              checkPopupTitle,
+              expectedTitle.toLowerCase(Locale.GERMAN),
+              "Wrong popup title for Pick or create a case");
+          softly.assertAll();
+          webDriverHelpers.clickOnWebElementBySelector(PICK_A_EXISTING_CASE_LABEL_DE);
+        });
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:304]: same, method name is general, but code logic covers DE. Can we make a generic method to cover both en and de?

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java [hunk 7]
```diff
@@ -226,6 +259,58 @@ public CreateNewTravelEntrySteps(
...
+
+    When(
+        "^I check if created travel entries are listed in the epidemiological data tab$",
+        () -> {
+          webDriverHelpers.isElementDisplayedIn20SecondsOrThrowException(
+              By.xpath("//div[text()='" + entryPoint + "']"));
```
Comment (rdutu-vg) [inline sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:310]: we should take this value: entryPoint  from a POJO, so we'll make the code reusable

**Review by rdutu-vg — CHANGES_REQUESTED** [review]

File: sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/entries/CreateNewTravelEntryPage.java (hunk 1 above)
Comment (pk-sgent) [reply sormas-e2e-tests/src/main/java/org/sormas/e2etests/pages/application/entries/CreateNewTravelEntryPage.java:47]: That kind of string we don't have in EN version: W\u00E4hlen Sie eine passende Person

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/TravelEntryService.java (hunk 2 above)
Comment (pk-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/entities/services/TravelEntryService.java:66]: good point

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java (hunk 4 above)
Comment (pk-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:252]: i rename it to 'person'

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java (hunk 5 above)
Comment (pk-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:278]: I can't :( As you can see i need to check if German label is displayed
` String expectedTitle = "Person ausw\u00E4hlen oder erstellen";`

File: sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java (hunk 6 above)
Comment (pk-sgent) [reply sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/entries/CreateNewTravelEntrySteps.java:304]: we still have no scenario for en version, if will be, i don't know who will be do that and what kind of idea will have for that, in my opinion if somebody wanna reuse it then, no problem to change by him and make directly with need

**Review by rdutu-vg — APPROVED** [review]

**Pull request merged**